### PR TITLE
Disable some hyper-specific NSURLCache tests on OS X. Fixes #773.

### LINKED ISSUE #773: NSURLCache tests fail on OSX
[----------] 3 tests from NSURLCache
[ RUN      ] NSURLCache.StorageAndRetrieval
Foundation/NSURLCacheTests.mm:45: Failure
Expected: (nullptr) != (oneComCached), actual: NULL vs NULL
[  FAILED  ] NSURLCache.StorageAndRetrieval (1 ms)
[ RUN      ] NSURLCache.Eviction
Foundation/NSURLCacheTests.mm:54: Failure
Value of: [cache currentMemoryUsage]
  Actual: 0
Expected: 13
[  FAILED  ] NSURLCache.Eviction (0 ms)
[ RUN      ] NSURLCache.Promotion
Foundation/NSURLCacheTests.mm:79: Failure
Expected: (nullptr) != ([cache cachedResponseForRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://two.com"]]]), actual: NULL vs NULL
[  FAILED  ] NSURLCache.Promotion (1 ms)
[----------] 3 tests from NSURLCache (4 ms total)

### PR DESCRIPTION
**Pull request by DHowett-MSFT**

### CODE REVIEW

File: tests/unittests/Foundation/NSURLCacheTests.mm [hunk 1]
```diff
@@ -63,7 +66,7 @@ static void _addFakeCacheResponse(NSURLCache* cache, NSCachedURLResponse* respon
     EXPECT_EQ(0, [cache currentMemoryUsage]);
 }
 
-TEST(NSURLCache, Promotion) {
+OSX_DISABLED_TEST(NSURLCache, Promotion) {
```
Comment (MSFTFox) [inline tests/unittests/Foundation/NSURLCacheTests.mm]: >OSX_DISABLED_TEST [](start = 0, length = 17)

nit: provide similar comment as above or just mention that this test behaves similarly.

**Comment by MSFTFox** [comment]
:shipit:

**Comment by ms-jihua** [comment]
:shipit:

**Comment by msft-Jeyaram** [comment]
:shipit:

File: tests/unittests/Foundation/NSURLCacheTests.mm (hunk 1 above)
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSURLCacheTests.mm]: the above comment covers both :)

**Pull request merged**